core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -0,0 +1,660 @@
+    ) {
+        let thread_id = self.in_flight_tracker.complete_batch(batch_id);
+        for transaction in transactions {
+            let account_locks = transaction.get_account_locks_unchecked();
+            self.account_locks.unlock_accounts(
+                account_locks.writable.into_iter(),
+                account_locks.readonly.into_iter(),
+                thread_id,
+            );
+        }
+    }
+
+    /// Send all batches of transactions to the worker threads.
+    /// Returns the number of transactions sent.
+    fn send_batches(&mut self, batches: &mut Batches) -> Result<usize, SchedulerError> {
+        (0..self.consume_work_senders.len())
+            .map(|thread_index| self.send_batch(batches, thread_index))
+            .sum()
+    }
+
+    /// Send a batch of transactions to the given thread's `ConsumeWork` channel.
+    /// Returns the number of transactions sent.
+    fn send_batch(
+        &mut self,
+        batches: &mut Batches,
+        thread_index: usize,
+    ) -> Result<usize, SchedulerError> {
+        if batches.ids[thread_index].is_empty() {
+            return Ok(0);
+        }
+
+        let (ids, transactions, max_age_slots, total_cus) = batches.take_batch(thread_index);
+
+        let batch_id = self
+            .in_flight_tracker
+            .track_batch(ids.len(), total_cus, thread_index);
+
+        let num_scheduled = ids.len();
+        let work = ConsumeWork {
+            batch_id,
+            ids,
+            transactions,
+            max_age_slots,
+        };
+        self.consume_work_senders[thread_index]
+            .send(work)
+            .map_err(|_| SchedulerError::DisconnectedSendChannel("consume work sender"))?;
+
+        Ok(num_scheduled)
+    }
+
+    /// Given the schedulable `thread_set`, select the thread with the least amount
+    /// of work queued up.
+    /// Currently, "work" is just defined as the number of transactions.
+    ///
+    /// If the `chain_thread` is available, this thread will be selected, regardless of
+    /// load-balancing.
+    ///
+    /// Panics if the `thread_set` is empty.
+    fn select_thread(
+        thread_set: ThreadSet,
+        chain_thread: Option<ThreadId>,
+        batches_per_thread: &[Vec<SanitizedTransaction>],
+        in_flight_per_thread: &[usize],
+    ) -> ThreadId {
+        if let Some(chain_thread) = chain_thread {
+            if thread_set.contains(chain_thread) {
+                return chain_thread;
+            }
+        }
+
+        thread_set
+            .contained_threads_iter()
+            .map(|thread_id| {
+                (
+                    thread_id,
+                    batches_per_thread[thread_id].len() + in_flight_per_thread[thread_id],
+                )
+            })
+            .min_by(|a, b| a.1.cmp(&b.1))
+            .map(|(thread_id, _)| thread_id)
+            .unwrap()
+    }
+
+    /// Gets accessed resources for use in `PrioGraph`.
+    fn get_transaction_resource_access(
```

Comment:
Yeah, sounds good. `ResourceKey` was the term I used in prio-graph since it's more generic than just accounts..but within context of solana scheduling it is just accounts.